### PR TITLE
Remove deprecated extract_along_coord

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: sunpy/map/tests/test_maputils.py [hunk 1]
```diff
@@ -277,38 +275,6 @@ def test_contains_coord(aia171_test_map):
...
-        aia171_test_map.unit
-    )
-    line_discrete_pix = aia171_test_map.world_to_pixel(line_discrete)
-    assert np.allclose(pix_diag[:, 0], line_discrete_pix.x.value)
-    assert np.allclose(pix_diag[:, 1], line_discrete_pix.y.value)
-    assert u.quantity.allclose(intensity_diag, intensity)
-
-
-@pytest.mark.filterwarnings("ignore:The extract_along_coord function is deprecated:sunpy.util.exceptions.SunpyDeprecationWarning")
-def test_extract_along_coord_one_point_exception(aia171_test_map):
```
Comment (hayesla) [inline sunpy/map/tests/test_maputils.py:298]: just looking at this, this should really be added as a test to `pixelate_coord_path`, and similarly with the test below. Maybe they can go in a different pr  though

Comment (nabobalis) [reply sunpy/map/tests/test_maputils.py:298]: If they go into a separate PR, we can backport it easier. Do we want to track this with an issue?

Comment (dstansby) [reply sunpy/map/tests/test_maputils.py:298]: Why would we need to backport anything here?

Comment (nabobalis) [reply sunpy/map/tests/test_maputils.py:298]: If we want to migrate these tests to the current API, I think those should be backported to ensure we don't break anything in the future.

Comment (hayesla) [reply sunpy/map/tests/test_maputils.py:298]: could they just be added in a new PR? (also the second test is something that is needed for the pixelate_coord_path I think (see #7205)

Comment (ayshih) [reply sunpy/map/tests/test_maputils.py:298]: Aside from the fix that is needed for #7205, these tests do not need to be preserved.

**Review by wtbarnes — APPROVED** [review]

**Review by Cadair — APPROVED** [review]
